### PR TITLE
Add support to get tagged jobs

### PR DESCRIPTION
**Pull request by dlecocq**
@b4hand @evanbattaglia @kq2

### CODE REVIEW

File: test/jobs.test.js [hunk 1]
```diff
@@ -66,4 +68,44 @@ describe('Jobs', () => {
...
+    });
+
+    it('gives us access to all the jobs', () => {
+      return client.jobs.tagged(tag)
+        .get('jobs')
+        .then(jobs => expect(new Set(jobs)).to.eql(new Set(jids)));
+    });
+  });
+
+  describe('get', () => {
```
Comment (b4hand) [inline test/jobs.test.js]: Was this just untested previously?

Comment (b4hand) [reply test/jobs.test.js]: Never mind, it's new too.

**Comment by b4hand** [comment]
Looks like it is failing on older versions of node due to its use of `...`. I'm not sure whether it is worthwhile to maintain compatibility to Node 4 & 5 at this point or not.

**Comment by dlecocq** [comment]
Forgot about `...` in earlier versions of `node`. I'll address that.

**Comment by b4hand** [comment]
Node 4 is EOL at the end of April and Node 5 is already EOL according to [here](https://github.com/nodejs/Release)

**Comment by dlecocq** [comment]
Do you want to just drop 4 and 5 entirely? I have not qualms with that. We use 8

**Comment by b4hand** [comment]
We should probably add 8 & 9 to the list of tested versions. I'm fine dropping 4 & 5 support.

**Comment by dlecocq** [comment]
Rgr. PR incoming.

**Comment by dlecocq** [comment]
Ah. That's a good point. I'll do so now.

**Pull request merged**